### PR TITLE
[AAP-19475] Add Group Modal

### PR DESCRIPTION
**Pull request by prat98**
This PR is related to AAP-19475 which is about creating a pop-up modal to allow users associate a host with a single or multiple groups at once

### CODE REVIEW

File: frontend/awx/resources/inventories/inventoryHostsPage/InventoryHostGroupsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+export interface InventoryHostGroupsAddModalProps {
+  onAdd: (items: InventoryGroup[]) => void;
+}
+
+export function InventoryHostGroupsAddModal(props: { onAdd: (items: InventoryGroup[]) => void }) {
+  const params = useParams<{ id: string; inventory_type: string; host_id: string }>();
+
+  const toolbarFilters = useHostsGroupsFilters();
+  const tableColumns = useHostsGroupsColumns({ disableLinks: true });
```
Comment (skochay330) [inline frontend/awx/resources/inventories/inventoryHostsPage/InventoryHostGroupsModal.tsx:19]: The sorting arrow in the name column header seems to be disabled

**Review by skochay330 — COMMENTED** [review]
Just some minor tweaks needed, otherwise looks great!

File: frontend/awx/resources/inventories/inventoryHostsPage/InventoryHostGroupsModal.tsx (hunk 1 above)
Comment (prat98) [reply frontend/awx/resources/inventories/inventoryHostsPage/InventoryHostGroupsModal.tsx:19]: This seems to be a framework level issue with all components using `MultiSelectDialog`. The bug tracking this work can be found here https://issues.redhat.com/browse/AAP-20202

File: frontend/awx/resources/inventories/inventoryHostsPage/hooks/useAddInventoryGroups.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    disableSort: true,
+  });
+  const addActionNameColumn = useNameColumn({ disableLinks: true, disableSort: true });
+  const actionColumns = useMemo(() => [addActionNameColumn], [addActionNameColumn]);
+  const bulkAction = useAwxBulkConfirmation<InventoryGroup>();
+  const params = useParams<{ id: string; inventory_type: string; host_id: string }>();
+  const addGroups = (groups: InventoryGroup[]) => {
+    bulkAction({
+      title: t('Add groups to inventory', { count: groups.length }),
+      confirmText: t('Yes, I confirm that I want to add these {{count}} groups.', {
```
[the remark this answers is not in the document]
Comment (prat98) [reply frontend/awx/resources/inventories/inventoryHostsPage/hooks/useAddInventoryGroups.ts]: Changed to action dialog to match design pattern of the rest of the UI. There is no confirm text anymore

**Review by skochay330 — APPROVED** [review]

**Pull request merged**